Tests/Realm.Tests/Database/PrimaryKeyTests.cs

```diff
@@ -192,6 +192,46 @@ private RealmObjectBase FindByPKGeneric(Type type, object primaryKeyValue, PKTyp
             return (RealmObjectBase)genericMethod.MakeGenericMethod(type).Invoke(_realm, new[] { primaryKeyValue });
         }
 
+        [Test]
+        public void ExceptionIfMismatchingPKType()
```

Comment:
We try to avoid writing overly broad tests that cover a bunch of parameters. A better approach would be to split these into smaller tests and extract a core test that does the verifications:

```csharp
private readonly IEnumerable<dynamic> _primaryKeyValues = new[] { "42", 123, ObjectId.GenerateNewId(), Guid.NewGuid() };

[Test]
public void RealmFind_WhenPKIsIntAndArgumentIsNot_Throws() => RealmFind_IncorrectPKArgument_Throws<IntPrimaryKeyWithValueObject>();

// Rest of PK types.

private void RealmFind_IncorrectPKArgument_Throws<T>()
{
    foreach (var pk in _primaryKeyValues)
    {
        // Skip the PK if its type is equal to the type of the object
        Assert.That(() => _realm.Find<T>(pk), Throws...);

        // Also add a test for the dynamic API: _realm.DynamicApi.Find("")
    }
}
```